File: server/strategies/substitution.py
from .cipher_interface import CipherInterface

class SubstitutionCipher(CipherInterface):
    def decrypt(self, text: str, key: str) -> str:
        if len(key) != 26: return "Hata: Anahtar 26 harfli bir alfabe olmalı."
        
        std_alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        key_map = {k.upper(): v for k, v in zip(key, std_alphabet)}
        
        result = ""
        for char in text:
            if char.isalpha():
                decoded = key_map.get(char.upper(), char)
                result += decoded if char.isupper() else decoded.lower()
            else:
                result += char
        return result

File: lib/ciphers/substitution.dart
import 'cipher_base.dart';

class SubstitutionCipher implements CipherBase {
  @override
  String get name => "Substitution (Yerine Koyma)";
  @override
  String get id => "substitution";

  @override
  String encrypt(String text, String key) {
    if (key.length != 26) return "Anahtar 26 harfli bir alfabe olmalı.";

    String std = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    String upperKey = key.toUpperCase();
    StringBuffer result = StringBuffer();

    for (int i = 0; i < text.length; i++) {
      String char = text[i].toUpperCase();
      int index = std.indexOf(char);
      if (index != -1) {
        String enc = upperKey[index];
        result.write(text[i] == char ? enc : enc.toLowerCase());
      } else {
        result.write(text[i]);
      }
    }
    return result.toString();
  }
}